refactor(Navigation): centralize BottomNavigationMenu in MainActivity

cldylan opened the pull request:
## Summary
This PR refactors the bottom navigation implementation by moving the `BottomNavigationMenu` out of individual screens and managing it centrally in `MainActivity`. This is mainly to fix an issue when quickly moving between tags, which used to recompose the whole screen making the bottom bar temporarily disappear.

## Changes
- Moved `BottomNavigationMenu` from `CurrentTripScreen`, `ProfileScreen`, `MyTripsScreen`, and `PastTripsScreen` into a single `Scaffold` in `MainActivity`.
- Added logic in `MainActivity` to conditionally display the bottom bar only for primary navigation routes (`CurrentTrip`, `MyTrips`, `Profile`).
- Encapsulated the `NavHost` within the `Scaffold` to ensure proper layout and consistent UI behavior.
- Simplified individual screens by removing bottom bar code, improving readability and maintainability.
- Overall, this refactor makes the navigation structure more robust and easier to extend in the future.

Vincentdfvh commented on app/src/main/java/com/github/swent/swisstravel/ui/mytrips/MyTripsScreen.kt:
```diff
@@ -172,12 +169,6 @@ fun MyTripsScreen(
-        BottomNavigationMenu(
-            selectedTab = Tab.MyTrips,
-            onTabSelected = { tab -> navigationActions?.navigateTo(tab.destination) },
-            modifier = Modifier.testTag(NavigationTestTags.BOTTOM_NAVIGATION_MENU))
```
## Comment
LGTM

Vincentdfvh reviewed (approved):
## Change

The PR seems well done. You created the scaffold in mainActivity, allowed the bottomBar to be shown when in the 3 main screens (myTrips, profile and current trip), and moved the composables of mainActivity in this scaffold. You also deleted the repetitive bottomBar in the three files corresponding to MyTrips, Current Trip and Profile.

The only thing missing is the line coverage, only at 68%. I believe it should be 80%, maybe add a little test that checks that the bottomBar is Displayed when the user is in the MytripsScreen, CurrentTripScreen, and ProfileScreen.

cldylan commented:
> ## Change
> 
> The PR seems well done. You created the scaffold in mainActivity, allowed the bottomBar to be shown when in the 3 main screens (myTrips, profile and current trip), and moved the composables of mainActivity in this scaffold. You also deleted the repetitive bottomBar in the three files corresponding to MyTrips, Current Trip and Profile.
> 
> The only thing missing is the line coverage, only at 68%. I believe it should be 80%, maybe add a little test that checks that the bottomBar is Displayed when the user is in the MytripsScreen, CurrentTripScreen, and ProfileScreen.

Thanks! The tests you suggest actually already exist. Since I refactored the code, they still work. I think it is just low due to it being in `MainActivity.kt`. I am hoping that the line coverage goes up when we make the end to end tests!

cldylan commented on app/src/main/java/com/github/swent/swisstravel/MainActivity.kt:
```diff
@@ -133,182 +147,191 @@ fun SwissTravelApp(
+                          tripId = tripId,
+                          onBack = { navController.popBackStack() },
+                          onSaved = { navController.popBackStack() },
+                          onDelete = { navigationActions.navigateTo(Screen.MyTrips) })
```
Yep!
